mpa/modules/models/segmentors/mean_teacher_segmentor.py

```diff
@@ -0,0 +1,102 @@
+from mmseg.models import SEGMENTORS, build_segmentor
+from mmseg.models.segmentors.base import BaseSegmentor
+from mmseg.ops import resize
+import torch
+import functools
+from collections import OrderedDict
+from mpa.utils.logger import get_logger
+logger = get_logger()
+
+
+@SEGMENTORS.register_module()
+class MeanTeacherNaive(BaseSegmentor):
+    def __init__(self, orig_type=None, unsup_weight=0.1, warmup_start_iter=30, **kwargs):
+        logger.info('MeanTeacherNaive Segmentor init!')
+        super(MeanTeacherNaive, self).__init__()
+        self.test_cfg = kwargs['test_cfg']
+        self.warmup_start_iter = warmup_start_iter
+        self.count_iter = 0
+
+        cfg = kwargs.copy()
+        if orig_type == 'EncoderDecoder':
+            cfg['type'] = 'EncoderDecoder'
+            self.align_corners = cfg['decode_head'].align_corners
+        self.model_s = build_segmentor(cfg)
+        self.model_t = build_segmentor(cfg)
+        self.unsup_weight = unsup_weight
+
+        # Hooks for super_type transparent weight load/save
+        self._register_state_dict_hook(self.state_dict_hook)
+        self._register_load_state_dict_pre_hook(
+            functools.partial(self.load_state_dict_pre_hook, self)
+        )
+
+    def extract_feat(self, imgs):
+        return self.model_s.extract_feat(imgs)
```

Comment:
Could you check teacher inference? Unbiased teacher on detection uses teacher model to extract_feat and test.

Reply:
Semi-seg are all inferencing with student models. Does it need to match with detection task?

Reply:
You don't have to match but it's suggestion. As far as I know, in academia, teacher model inferences when evaluate the model.